vllm/compilation/fusion.py

```diff
@@ -75,6 +98,16 @@ def record_match_fn(match: Match):
     register_replacement(rms_pattern_residual_static, rms_replacement_residual_static, inputs, fwd_only, my_patterns,
                          extra_check=record_match_fn)
 
+    # silu-mul quant
+    x = torch.empty((128, 256), device="cuda", dtype=torch.float16)
+    scale = torch.empty((1,1), device="cuda" , dtype=torch.float32)
```

Comment:
Do we need to set device/type here?  I guess it doesn't matter a whole lot since I think these are mostly ignored

Reply:
This was just sage's diff I added. I would not land this without fully removing the allocations from patterns